### ISSUE TITLE
Unify ttnn.pad interface with torch.nn.functional.pad

### ISSUE THREAD

**Issue by philei-tt** [tenstorrent/tt-metal#21198]
### Is your feature request related to a problem? Please describe

In pytorch2-ttnn compiler there are a few ugly places where I have to convert from/to ttnn padding format. 

### Describe the solution you'd like

ttnn, for some reason, supports two padding formats, both differ from the one used in PyTorch: 
- pad=[output_padded_shape] + [input_tensor_start (i.e. left padding for each dim)]
- List[Tuple[int, int]] - [(pad_left, pad_right), ...]

While [torch's pad](https://pytorch.org/docs/stable/generated/torch.nn.functional.pad.html) has basically flattened second variant. 

I think this is a half-hour PR. We just have to add another overload for bindings, where we use some one-liner to convert to second supported padding format + update docs. 

Also, it would be great to clean up the code to have only one unified format inside the ttnn code, but this might take more time.

**Comment by philei-tt** [comment]
@ayerofieiev-tt @ntarafdar please take a look. If you're okay with adding another overload, I can quickly open PR for that
